transportccextension.go

```diff
@@ -0,0 +1,49 @@
+package rtp
+
+import (
+	"encoding/binary"
+	"errors"
+)
+
+const (
+	// TransportCCExtensionSize 4 byte
+	TransportCCExtensionSize = 4
+)
+
+var (
+	errTooSmall = errors.New("buffer too small")
+)
+
+// TransportCCExtension is a extension payload format in
+// https://tools.ietf.org/html/draft-holmer-rmcat-transport-wide-cc-extensions-01
+// 0                   1                   2                   3
+// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |       0xBE    |    0xDE       |           length=1            |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |  ID   | L=1   |transport-wide sequence number | zero padding  |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+type TransportCCExtension struct {
+	ID  uint8
+	L   uint8
```

Comment:
Since L will always equal 1 do you need to set it here?

You could just assert that it's 1 on Unmarshal instead. That will make constructing a TransportCCExtenstion struct easier since you won't need to remember to set L: 1

Reply:
`L` is a `(length-1)` of the element, in this case two bytes.
`L=3` version of Transport CC extension with extra fields looks been proposed.
https://webrtc.org/experiments/rtp-hdrext/transport-wide-cc-02/
```
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |  ID   | L=3   |transport-wide sequence number |T|  seq count  |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |seq count cont.|
 +-+-+-+-+-+-+-+-+
```

In my understanding, rfc5285 defines just a way to transmit some pairs of ID and value which are "defined by profile".
So, I'd like to suggest to make it something general like `HeaderExtension` and marshal/unmarshal between binary and `map[ID][]byte`.
How do you think?

Reply:
I've done some thinking along those lines here: https://github.com/pion/rtp/issues/17

Sadly I don't have time to implement it.

I figured we could add that later in a backwards compatible way but do this for now since adam needs this extension immediately.

Reply:
> Looks nice to have header extensions parser!
> I roughly read rfc5285 and added a comment [#30 (comment)](https://github.com/pion/rtp/pull/30#discussion_r363129755).
> Please let me know if my understanding is wrong :)

New RFC:
https://tools.ietf.org/html/rfc8285